lambda/tests/test_new_account_iam_role.py

```diff
@@ -0,0 +1,315 @@
+"""Test event handler and main() of new_account_iam_role.
+
+This is testing a rather basic lambda function, so the tests are
+basic as well:
+
+    - test arguments to main()
+    - test some of the functions invoked by main()
+    - test event handler arguments
+"""
+import json
+import os
+
+import pytest
+from moto import mock_iam
+from moto import mock_sts
+from moto.core import ACCOUNT_ID
+import boto3
+
+import new_account_iam_role as lambda_func
+
+AWS_REGION = os.getenv("AWS_REGION", default="aws-global")
+
+
+@pytest.fixture
+def lambda_context():
+    """Create mocked lambda context injected by the powertools logger."""
+
+    class LambdaContext:  # pylint: disable=too-few-public-methods
+        """Mock lambda context."""
+
+        def __init__(self):
+            """Initialize context variables."""
+            self.function_name = "test"
+            self.memory_limit_in_mb = 128
+            self.invoked_function_arn = (
+                f"arn:aws:lambda:{AWS_REGION}:{ACCOUNT_ID}:function:test"
+            )
+            self.aws_request_id = "52fdfc07-2182-154f-163f-5f0f9a621d72"
```

Comment:
Is this value significant? Looks like a UUID. Should we just generate one?

```suggestion
            self.aws_request_id = str(uuid.uuid4())
```

Reply:
Done